src/chainstate/stacks/miner.rs

```diff
@@ -125,6 +132,224 @@ impl From<&UnconfirmedState> for MicroblockMinerRuntime {
+    pub error: String,
+}
+
+/// Represents an event for a transaction that was skipped, but might succeed later.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct TransactionSkippedEvent {
+    pub txid: Txid,
+    pub reason: String,
+}
+
+/// `TransactionResult` represents the outcome of transaction processing.
+/// We use this enum to involve the compiler in forcing us to always clearly
+/// indicate the outcome of a transaction.
+///
+/// There are currently three outcomes for a transaction:
+/// 1) succeed
+/// 2) fail, may be tried again later
+/// 3) be skipped for now, to be tried again later
+#[derive(Debug)]
+pub enum TransactionResult {
+    /// Transaction has already succeeded.
+    Success(TransactionSuccess),
+    /// Transaction failed. It may succeed later depending on the error.
+    ProcessingError(TransactionError),
+    /// Transaction wasn't ready to be be processed, but might succeed later.
+    /// The bool represents whether mempool propagation should halt or continue
+    Skipped(bool, TransactionSkipped),
+}
+
+/// This struct is used to transmit data about transaction results through either the `mined_block`
+/// or `mined_microblock` event.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum TransactionEvent {
+    /// Transaction has already succeeded.
+    Success(TransactionSuccessEvent),
+    /// Transaction failed. It may succeed later depending on the error.
+    ProcessingError(TransactionErrorEvent),
+    /// Transaction wasn't ready to be be processed, but might succeed later.
+    /// The bool represents whether mempool propagation should halt or continue
+    Skipped(TransactionSkippedEvent),
+}
+
+impl TransactionResult {
+    /// Logs a queryable message for the case where `txid` has succeeded.
+    pub fn log_transaction_success(tx: &StacksTransaction) {
+        let tx_id_str = format!("{}", &tx.txid());
+        info!("Tx successfully processed.";
+            "event_name" => %"transaction_result",
+            "tx_id" => %tx_id_str,
+            "event_type" => %"success",
+        );
+    }
+
+    /// Logs a queryable message for the case where `txid` has failed
+    /// with error `err`.
+    pub fn log_transaction_error(tx: &StacksTransaction, err: &Error) {
+        let tx_id_str = format!("{}", &tx.txid());
+        warn!("Tx processing failed with error: {}", err;
+            "event_name" => %"transaction_result",
+            "tx_id" => %tx_id_str,
+            "event_type" => %"error",
+        );
+    }
+
+    /// Logs a queryable message for the case where `tx` has been skipped
+    /// for reason `reason`.
+    pub fn log_transaction_skipped(tx: &StacksTransaction, reason: String) {
+        let tx_id_str = format!("{}", &tx.txid());
+        info!(
+            "Tx processing skipped for reason: {}", reason;
+            "event_name" => %"transaction_result",
+            "tx_id" => %tx_id_str,
+            "event_type" => %"skip"
+        );
```

Comment:
Here's a suggested change for these lines:

```suggestion
        info!(
            "Tx processing skipped";
            "event_name" => "transaction_result",
            "tx_id" => %tx.txid(),
            "event_type" => "skip",
            "reason" => reason,
        );
```

Basically, make "reason" a field in the structured log. Also, the `%` prefix just tells the macro to invoke the `Display` format implementation to write the passed object as a str, which the str arguments don't need.